Assets/MixedRealityToolkit/Interfaces/InputSystem/IMixedRealityInputSystem.cs

```diff
@@ -52,6 +52,11 @@ public interface IMixedRealityInputSystem : IMixedRealityEventSystem
         /// </summary>
         IMixedRealityEyeGazeProvider EyeGazeProvider { get; }
 
+        /// <summary>
+        /// Last pointer to raise a pointer down event. May be useful for choosing a primary pointer in two handed setups.
+        /// </summary>
+        IMixedRealityPointer LastPointerDown { get; }
```

Comment:
Please consider implementing this as LastFocusPointer instead of triggering on the InputDown event.

Reply:
The issues I see with using focus are that:
- You don't have a primary pointer until you actually point at something that can receive focus
- The primary pointer could change while you're actively using it. E.g. you're dragging something with your right hand and accidentally point at a focusable object with your left.
I'm currently prototyping something along the lines of using the first active pointer in the last controller detected or gaze if none.

The difficulties we're having an coming up with a good criteria for primary pointer makes me think that, besides having some sort of default, we need to make sure that that logic can be customized easily.

Reply:
> The difficulties we're having an coming up with a good criteria for primary pointer makes me think that, besides having some sort of default, we need to make sure that that logic can be customized easily.

I totally agree @luis-valverde-ms. Maybe we should spin off a short design call and invite anyone that wants to join to discuss? I'd be more than happy to participate.

Reply:
@jbienzms, good idea. I would be most interested in what _primary pointer_ means for people, the use cases they have in mind for it and how they would like the API to get access to it to look like. What would be the right place to schedule that call?

Reply:
Great question @luis-valverde-ms . @davidkline-ms and @Yoyozilla, is there a process for spinning up community discussion calls around a feature like this? We can of course use time in the ship room if there's time available, but it might be a good idea to have a process for one-off collaborative design discussion calls around specific subjects where community members might want to participate.

Reply:
So on the topic of determining the best pointer, I've been doing some tests, and I found a good criteria.

Basically, just grabbing the UIRayCastCamera's forward vector, and then checking the dot product against forward direction of all the pointers in the FocusProvider. The dot product with the highest value (closest to 1), seems to best represent the pointer I'm looking for.

However, with this in mind there are a few caveats. You need to check if the pointer you're seeking is null every frame - this requires casting to monobehavior in order to get the unity overridden null check.

Moreover, the camera gaze pointer seems to be set to null every single frame when no hand is present in view. This seems to be a bug with handling of the gaze pointer.

Reply:
Thanks @provencher , do you mind opening a bug with repro instructions for that issue with the gaze pointer?

It sounds like a method based on the pointer angle with the forward camera direction could be a useful option for some use cases. I'm concerned about two handed manipulation scenarios where you can potentially switch main pointer in the middle of a manipulation, but I think we could/should support the forward method as an option depending on your findings.

Let me know how it goes in your use case and if you find any more caveats.

Reply:
I put some code forward for my method
https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/3957#issuecomment-488834026